sdk/cosmosdb/cosmos/src/queryExecutionContext/hybridQueryExecutionContext.ts

```diff
@@ -455,8 +456,14 @@ export class HybridQueryExecutionContext implements ExecutionContext {
     globalStats: GlobalStatistics,
   ): QueryInfo[] {
     return componentQueryInfos.map((queryInfo) => {
-      if (!queryInfo.hasNonStreamingOrderBy) {
-        throw new Error("The component query must have a non-streaming order by clause.");
+      let rewrittenOrderByExpressions = queryInfo.orderByExpressions;
+      if (queryInfo.orderBy && queryInfo.orderBy.length > 0) {
```

Comment:
why have we introduced this check?

Reply:
In optimized query plan, orderBy is not present for RRF queries.